### PR TITLE
Determining mixed-layer depth from relative humidity peak

### PR DESCRIPTION
**Pull request by leifdenby**
I've started implementing https://github.com/eurec4a/eurec4a-environment/issues/12 to have a calculation for `h_maxRH`.

Here's what I've got so far:

mixed_layer_heights__z_RHmax

I need to work out if these numbers look reasonable. What do you think @annalea-albright?
Also, I need to work out why the time dimension isn't bringing in the actual time values (datatime) instead of what looks like time-index

### CODE REVIEW

**Comment by annalea-albright** [comment]
I'll have a look at your implementation tomorrow morning! For individual sondes, it seems like the results should be more sensitive to whether or not you specify an upper and lower boundary condition for the mixed layer height (i.e. search from 400-900m). For circle-means and larger, the profiles are smoother, so, in my experience, this doesn't seem to make such a difference. For individual sondes, I had limited the BL height to be between 400-900m, and then the center of mass is around 600m, which seems higher than in the figure above, but you do have the same 'bunching' at the upper limit (seems like 700m in yours, or 900m in mine) ... if anyone has an idea how to think about these choices, that would be great. My 400m lower limit might actually be too high for individual sondes, because a given sonde could, for instance, fall in a cold pool, whose BL is observed to only be well-mixed for ~200m, I think 
image

**Comment by annalea-albright** [comment]
image
here's the distribution from all Joanne sondes if you set the lower limit to be 100m ... but anyway, I'll look at your implementation first :)

**Comment by leifdenby** [comment]
Thanks @annalea-albright! You're right, it is strange that I don't see any that are at 900m. Also I should change the lower-limit to be the same as you had before. And I'll make a vertical distribution plot like you have too.

File: eurec4a_environment/variables/boundary_layer/mixed_layer_height.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    """
+    ds_mixed_layer = ds.sel({altitude: slice(z_min, z_max)})
+
+    da_rh = ds_mixed_layer[rh]
+    dims = list(ds.dims.keys())
+    # drop the height coord
+    del dims[dims.index(altitude)]
+
+    peakRH_idx = da_rh.argmax(dim=altitude)
+    h_peakRH = ds[altitude].isel({altitude: peakRH_idx})
```
Comment (annalea-albright) [inline eurec4a_environment/variables/boundary_layer/mixed_layer_height.py:16]: Do you have to add index of z_min=20 to peakRH_idx? For the results, I see heights that range from 20-700m

Comment (leifdenby) [reply eurec4a_environment/variables/boundary_layer/mixed_layer_height.py]: You're right that the indexing was wrong. I was picking by number into an array where I'd already sliced off values above and below 😄

**Comment by leifdenby** [comment]
I think it's working now @annalea-albright 😀 , here are the first 120 sondes from HALO:

Screenshot 2020-07-29 at 15 18 25

I've added my notebook to the commit too: https://github.com/leifdenby/eurec4a-environment/blob/mixed-layer-height-maxRH/notebooks/calculating%20mixed-layer%20depth%20from%20RH%20peak.ipynb

**Comment by Geet-George** [comment]
Can h_maxRH be plotted as scatter points instead of a line plot? A line would indicate continuity, which isn't the case here.

**Comment by leifdenby** [comment]
> Can h_maxRH be plotted as scatter points instead of a line plot? A line would indicate continuity, which isn't the case here.

Yes, good point. This is just a quick-hack plot to indicate that I think it's working. The actual plotting functions @JuleRadtke and @annalea-albright are working on

**Comment by Geet-George** [comment]
Ahh.. Sorry then. I thought this was the plotting PR. Getting super confused now...

**Pull request merged**